freshmaker/pyxis.py

```diff
@@ -215,6 +215,135 @@ def get_images_by_digests(self, digests):
+
+        # For every repo add it's auto_rebuild_tags info
+        for repo in repos:
+            reg_repo_pair = (repo['registry'], repo['repository'])
+            # one of repos isn't in previously constructed map,
+            # so there is inconsistency
+            if reg_repo_pair not in reg_repo_info:
+                log.warning("There is inconsistency in naming for: %s/%s",
+                            reg_repo_pair[0], reg_repo_pair[1])
+                continue
+            tags = repo.get('auto_rebuild_tags')
+            # If the repository doesn't have 'auto_rebuild_tags', don't proceed with it
+            if tags:
+                reg_repo_info[reg_repo_pair]['tags'] = set(tags)
+            else:
+                del reg_repo_info[reg_repo_pair]
+
+    def _filter_auto_rebuild_repos(self, reg_repo_info):
+        """
+        For every pair in mapping get history of every auto_rebuild tag with
+        filter of all nvrs from that registry/repo pair.
+
+        :param dict reg_repo_info: map of pairs (registry, repository) to a
+            dict containing nvrs of bundle images from that repo and
+            auto_rebuild tags of that repo
+        :return: pairs of (registry, repository) of every repo with images ever
+            published to a repo with auto_rebuild tag + info about it
+        :rtype dict(tuple(str, str): dict()):
+        """
+        params = {'include': 'data.brew.build'}
+        ret_repos = dict()
+        for reg_repo_pair, info in reg_repo_info.items():
+            if not info.get('nvrs'):
+                continue
+            fltr = f"brew.build=in=({','.join(info['nvrs'])})"
+            params["filter"] = fltr
+            reg, repo = reg_repo_pair
+            for tag in info.get('tags', []):
+                tag_history = \
+                    self._pagination(f"repositories/registry/{reg}/"
+                                     f"repository/{repo}/tag/{tag}", params)
+                # If there is at least one record with nvr(brew.build) of the
+                # image, it means that image was published to a repo with
+                # auto_rebuild tag
+                if tag_history:
+                    ret_repos[reg_repo_pair] = reg_repo_info[reg_repo_pair]
+                    break
+        return ret_repos
+
+    def get_auto_rebuild_taged_images(self, bundle_images):
+        """
+        Determine which bundle images are part of container repository and tag
+        that has auto-rebuilds enabled
+
+        :param list bundle_images: Images of operator bundles that should
+            be filtered
+        :return: Images that where published to repository with auto_rebuild tag
+        :rtype: list
+        """
+        reg_repo_info = {}
+        ret_images = []
+        for bundle_image in bundle_images:
+            bundle_nvr = bundle_image.get('brew', {}).get('build')
+            if not bundle_nvr:
+                log.warning("One of bundle images doesn't have brew.build")
```

Comment:
Is there an identifier that you can use in the log message to denote which bundle image doesn't have an NVR?

Reply:
For now I am querying Pyxis only for two fields and they are **data.brew** and **data.repositories**. So this check is more for preventing someone calling it with incorrect data structures. I could use **brew.nvra** to print that warning, but it doesn't have much sense since it's almost the same data that is missing.